docs/contributor/testing-strategy.md

```diff
@@ -0,0 +1,95 @@
+
+1. Business Logic
+
+    Includes the technical frameworks (e.g. Kubebuilder) and custom Golang code.
+
+2. Business Features
+
+    Combines the business logic into feature which is consumed by our customers.
+
+3. Product Integration
+
+    Verifies how our product is integarted into the technical landscape, how it interacts with 3rd party systems and how it is accessible by customers or remote systems.
+ 
+For each layer is a dedicated testing approach used:
+
+1. **Unit Testing for Business Logic:** Writing and executing tests for individual functions, methods, and components to verify their behavior and correctness in isolation.
+2. **Integration Testing for Business Features:** Validating the integration and interaction between different components, modules, and services in the project.
+3. **End-to-End Testing:** Testing the application as a whole in a production-like environment, mimicking real-world scenarios to ensure the entire system functions correctly, is performing well and secure.
+
+
+## Testing Approach
+
+### Unit Testing
+1. Identify critical functions, methods, and components that require testing.
+2. Write unit tests using GoUnit tests, Ginkgo and Gomega frameworks.
+3. Ensure tests cover various scenarios, edge cases, and possible failure scenarios. We try to verify business relevant logic with at least 65% code coverage.
+4. Test for both positive and negative inputs to validate the expected behavior.
+5. Mock external dependencies and use stubs or fakes to isolate the unit under test.
+6. Run unit tests periodically during development and before each PR to prevent regressions.
+7. Unit tests have to be executed as fast as possible to minimize roundtrip times. Long running tests should be excluded from requently executed test runs and be triggered periodically (e.g. 4 times a day)
+
+### Integration Testing
+1. The PO together with the team create an registry of implemented business features and define for each feature a suitable test scenario.
+2. Create a separate test suite for integration testing.
+3. Each test scenario is implemented in a separte test case. Use the Kubebuilder Test Framework and others to create test cases that interact with the Kubernetes cluster.  
+4. Test the interaction and integration of your custom resources, controllers, and other components with the Kubernetes API.
+5. Ensure test cases cover various aspects such as resource creation, updating, deletion, and handling of edge cases.
+6. Validate the correctness of event handling, reconciliation, and other control logic.
+7. Unit tests have to be executed as fast as possible to minimize roundtrip times and be applied for each PR. Long running tests should be excluded from requently executed test runs and be triggered periodically (e.g. 4 times a day)
+
+### End-to-End Testing
+1. Use Helm to create, deploy, and manage test clusters and environments that closely resemble the productive execution context.
```

Comment:
I understand that Helm can make more sense over Kustomize as this is a tool of choice of SRE. But it would be good if we aligned on that in the team.